backend/nfconfig/session_management_test.go

```diff
@@ -0,0 +1,221 @@
+		t.Fatalf("expected 1 session config, got %d", len(cfg.sessionManagement))
+	}
+	if cfg.sessionManagement[0].Upf.GetHostname() != "" {
+		t.Errorf("expected UPF to be unset, got %+v", cfg.sessionManagement[0].GetUpf())
+	}
+}
+
+func TestSyncSessionManagement_InvalidUpfHostname(t *testing.T) {
+	// Invalid UPF hostname. The session management gets a config without UPF.
+	slice := configmodels.Slice{
+		SliceName:       "bad-upf",
+		SliceId:         configmodels.SliceSliceId{Sst: "1", Sd: "010203"},
+		SiteDeviceGroup: []string{"dg-1"},
+		SiteInfo: configmodels.SliceSiteInfo{
+			Plmn: configmodels.SliceSiteInfoPlmn{Mcc: "001", Mnc: "01"},
+			Upf:  map[string]interface{}{"upf-name": 123}, // Invalid type
+		},
+	}
+
+	deviceGroupMap := map[string]configmodels.DeviceGroups{
+		"dg-1": {
+			IpDomainExpanded: configmodels.DeviceGroupsIpDomainExpanded{
+				Dnn: "internet", DnsPrimary: "8.8.8.8", UeIpPool: "10.1.1.0/24", Mtu: 1500,
+			},
+		},
+	}
+
+	cfg := &inMemoryConfig{}
+	cfg.syncSessionManagement([]configmodels.Slice{slice}, deviceGroupMap)
+
+	if len(cfg.sessionManagement) != 1 {
+		t.Fatalf("expected 1 session config, got %d", len(cfg.sessionManagement))
+	}
+	if cfg.sessionManagement[0].Upf.GetHostname() != "" {
+		t.Errorf("expected UPF to be unset, got %+v", cfg.sessionManagement[0].GetUpf())
+	}
+}
+
+func TestSyncSessionManagement_MissingDeviceGroup(t *testing.T) {
+	// Missing device group. The session management gets a config without IPDomain.
+	slice := configmodels.Slice{
+		SliceName:       "no-dg",
+		SliceId:         configmodels.SliceSliceId{Sst: "1", Sd: "010203"},
+		SiteDeviceGroup: []string{"dg-x"}, // not in deviceGroupMap
+		SiteInfo: configmodels.SliceSiteInfo{
+			Plmn: configmodels.SliceSiteInfoPlmn{Mcc: "001", Mnc: "01"},
+			Upf:  map[string]interface{}{"upf-name": "upf.local"},
+		},
+	}
+
+	cfg := &inMemoryConfig{}
+	cfg.syncSessionManagement([]configmodels.Slice{slice}, map[string]configmodels.DeviceGroups{})
+
+	if len(cfg.sessionManagement) != 1 {
+		t.Fatalf("expected 1 session config, got %d", len(cfg.sessionManagement))
+	}
+
+	if len(cfg.sessionManagement[0].GetIpDomain()) != 0 {
+		t.Errorf("expected no IP domains due to missing device group, got %+v", cfg.sessionManagement[0].GetIpDomain())
+	}
+}
+
+func TestSyncSessionManagement_SortedBySliceName(t *testing.T) {
+	// The session management gets a config sorted by slice name in ascending order.
+	slices := []configmodels.Slice{
+		{SliceName: "slice-b", SliceId: configmodels.SliceSliceId{Sst: "1", Sd: "010203"}, SiteInfo: configmodels.SliceSiteInfo{Plmn: configmodels.SliceSiteInfoPlmn{Mcc: "001", Mnc: "01"}, Upf: map[string]interface{}{"upf-name": "upf.local"}}},
+		{SliceName: "slice-a", SliceId: configmodels.SliceSliceId{Sst: "1", Sd: "010203"}, SiteInfo: configmodels.SliceSiteInfo{Plmn: configmodels.SliceSiteInfoPlmn{Mcc: "001", Mnc: "01"}, Upf: map[string]interface{}{"upf-name": "upf.local"}}},
+	}
+
+	cfg := &inMemoryConfig{}
```

Comment:
```suggestion
	cfg := inMemoryConfig{}
```
why pointers? this applies for the other tests